components/TextFieldGroup.tsx

```diff
@@ -0,0 +1,37 @@
+import { Flex, StyleProps, useMultiStyleConfig } from '@chakra-ui/react'
+import { Children, cloneElement, FC, isValidElement } from 'react'
+
+const TextFieldGroup: FC<StyleProps> = ({ children, ...props }) => {
+  const styles = useMultiStyleConfig('TextFieldGroup', props)
+  return (
+    <Flex
+      {...props}
+      sx={{
+        ...styles?.container,
+        '>:first-child': {
+          borderRightRadius: '0 !important',
+        },
+        '>:not(:first-child):not(:last-child)': {
+          borderRadius: '0',
+        },
+        '>:last-child': {
+          borderLeftRadius: '0 !important',
+        },
+      }}
+      alignItems="center"
+    >
+      {Children.toArray(children)
+        .filter(child => isValidElement(child))
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .map((child: any) => {
+          return cloneElement(child, {
```

Comment:
For my own edification, this is required because of how `@chakra-ui` components / styles are applied?